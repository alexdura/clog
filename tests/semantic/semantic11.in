P(0).

Q(3) :- BIND(x, y), BIND(y, z), BIND(z, x).

Q(2) :- BIND(x, y), BIND(y, z), BIND(z, t), P(t).

Q(1) :- BIND(x, x + x).

Q(0) :- BIND(x, x).