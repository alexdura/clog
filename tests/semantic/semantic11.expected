Free variable(s): [y] in the RHS of BIND(x, y).
Free variable(s): [z] in the RHS of BIND(y, z).
Free variable(s): [x] in the RHS of BIND(z, x).
Free variable(s): [x] in the RHS of BIND(x, (x)+(x)).
Free variable(s) [x] in (x)+(x) in rule Q(1) :- 
	BIND(x, (x)+(x)).

Free variable(s): [x] in the RHS of BIND(x, x).
